controller/src/main/java/org/jboss/as/controller/ModelControllerImpl.java

```diff
@@ -448,11 +449,19 @@ boolean boot(final List<ModelNode> bootList, final OperationMessageHandler handl
         EnumSet<OperationContextImpl.ContextFlag> contextFlags = rollbackOnRuntimeFailure
                 ? EnumSet.of(AbstractOperationContext.ContextFlag.ROLLBACK_ON_FAIL)
                 : EnumSet.noneOf(OperationContextImpl.ContextFlag.class);
+        //TODO Gigantic HACK to disable the runtime part of this for the core model testing.
+        //The core model testing currently uses RunningMode.ADMIN_ONLY, but in the real world
+        //the http interface needs to be enabled even when that happens.
+        //I don't want to wire up all the services unless I can avoid it, so for now the tests set this system property
+        //@see https://issues.jboss.org/browse/WFCORE-1509
+        EnumSet<OperationContextImpl.ContextFlag> firstBatchFlags = WildFlySecurityManager.getPropertyPrivileged("wildfly.test.boot.disable.rollback", null) == null
+                ? EnumSet.of(AbstractOperationContext.ContextFlag.ROLLBACK_ON_FAIL)
+                : EnumSet.noneOf(OperationContextImpl.ContextFlag.class);
```

Comment:
BTW, if all else is ok on this, ^^^ is thoughts for a future change, not required for this PR.